src/main/java/com/google/sps/servlets/addSchoolInfo.java

```diff
@@ -0,0 +1,179 @@
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/add-info")
+public class addSchoolInfo extends HttpServlet {
+
+  private Key existingSchoolKey;
+  private Key existingCourseKey;
+  private Key existingProfessorKey;
+
+  private Key newSchoolKey;
+  private Key newCourseKey;
+  private Key newProfessorKey;
+  private Key newTermKey;
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    PostRequestObject postRequestData = processPostRequest(request);
+    String schoolName = postRequestData.getSchool();
+    String courseName = postRequestData.getCourse();
+    String term = postRequestData.getTerm();
+    String units = postRequestData.getUnits();
+    String profName = postRequestData.getProf();
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Filter schoolFilter = new FilterPredicate("school-name", FilterOperator.EQUAL, schoolName);
+    Query schoolQ = new Query("School").setFilter(schoolFilter);
+    List<Entity> schoolQuery =
+        datastore.prepare(schoolQ).asList(FetchOptions.Builder.withDefaults());
+
+    if (schoolQuery.size() == 0) {
+      Entity newSchool = new Entity("School");
+      newSchool.setProperty("school-name", schoolName);
+      newSchoolKey = newSchool.getKey();
+      datastore.put(newSchool);
+    } else {
+      Entity existingSchool = schoolQuery.get(0);
+      existingSchoolKey = existingSchool.getKey();
+      newSchoolKey = null;
+    }
+
+    if (newSchoolKey != null) {
+      Entity newCourse = new Entity("Course", newSchoolKey);
+      newCourse.setProperty("course-name", courseName);
+      newCourse.setProperty("Units", units);
+      newCourseKey = newCourse.getKey();
+      datastore.put(newCourse);
+
+      Entity newProfessor;
+      newProfessor = new Entity("Professor", newSchoolKey);
+      newProfessor.setProperty("professor-name", profName);
+      newProfessorKey = newProfessor.getKey();
+      datastore.put(newProfessor);
+
+      Entity newTerm = new Entity("Term", newCourseKey);
+      newTerm.setProperty("term", term);
+      newTerm.setProperty("Professor", newProfessorKey);
+      datastore.put(newTerm);
+
+    } else {
+      Filter courseFilter = new FilterPredicate("course-name", FilterOperator.EQUAL, courseName);
+      Query courseQ = new Query("Course").setAncestor(existingSchoolKey).setFilter(courseFilter);
+      List<Entity> courseQuery =
+          datastore.prepare(courseQ).asList(FetchOptions.Builder.withDefaults());
+
+      // existing school --> new course
+      if (courseQuery.size() == 0) {
+        Entity newCourse;
+        if (existingSchoolKey != null) {
+          newCourse = new Entity("Course", existingSchoolKey);
+        } else {
+          newCourse = new Entity("Course", newSchoolKey);
+        }
+
+        newCourse.setProperty("course-name", courseName);
+        newCourse.setProperty("Units", units);
```

Comment:
maybe standardize this? so "Course Name" and "Units" or "course-name" and "units" so it's easier for Laura and I to use your database without having to manually refer to each property name